File: 0303-Num-Array/0303.go
type NumArray struct {
	sums []int
}

func Constructor(nums []int) NumArray {
	sums := make([]int, len(nums)+1)
	for i, v := range nums {
		sums[i+1] = sums[i] + v
	}
	return NumArray(sums)
}

func (na *NumArray) SumRange(i, j int) int {
	return na.sums[j+1] - ma.sums[i]
}

File: 301-400/0303-Num-Array/0303.py
class NumArray:
  def __init__(self, nums: List[int]):
    self.nums = [0]
    _sums = self.nums

    for num in nums:
      _sums.append(_sums[-1] + num)

  def sumRange(self, i: int, j: int) -> int:
    _sums =  self.nums
    return _sums[j + 1] - _sums[i]